venues/ICLR.cc/2018/Conference/python/reviewer-assignment-from-server.py

```diff
@@ -0,0 +1,88 @@
+'''
+this is a one-time script to run the assignment directly from the server (because it's too slow to run it over the network)
+'''
+
+## Import statements
+import argparse
+import sys
+import config
+import csv
+import openreview
+
+## Argument handling
+parser = argparse.ArgumentParser()
+parser.add_argument('noteid')
+parser.add_argument('tpms_file', help='should be iclr18_reviewer_assignments_1106(from_tpms).csv')
+parser.add_argument('--baseurl', help="base url")
+parser.add_argument('--username')
+parser.add_argument('--password')
+
+args = parser.parse_args()
+
+client = openreview.Client(baseurl=args.baseurl, username=args.username, password=args.password)
+
+submissions = client.get_notes(invitation=config.BLIND_SUBMISSION)
+for paper in submissions:
+    paper_num = str(paper.number)
+    paperinv = config.CONF + '/-/Paper' + paper_num
+    print("Adding groups for Paper"+ paper_num)
+    paperGroup = config.CONF + '/Paper' + paper_num
+    authorGroup = paperGroup + '/Authors'
+
+    ## Reviewer group - people that can see the review invitation
+    reviewerGroup = paperGroup + '/Reviewers'
+    client.post_group(openreview.Group(
+        id=reviewerGroup,
+        signatures=[config.CONF],
+        writers=[config.CONF],
+        members=[],
+        readers=[config.CONF, config.PROGRAM_CHAIRS, config.AREA_CHAIRS],
+        signatories=[]))
+
+    ## Area Chair group -
+    areachairGroup = paperGroup + '/Area_Chair'
+    client.post_group(openreview.Group(
+        id=areachairGroup,
+        signatures=[config.CONF],
+        writers=[config.CONF],
+        members=[],
+        readers=[config.CONF, config.PROGRAM_CHAIRS, config.AREA_CHAIRS, areachairGroup],
+        signatories=[areachairGroup]))
+
+
+# populate a configuration note with the TPMS assignments
+configuration_note = client.get_note(args.noteid)
+
+def tpms_assignment(configuration_note):
+    configuration_note.content['assignments'] = { 'Paper{0}'.format(n.number): {'assigned':[], 'forum':n.forum, 'title':n.content['title']} for n in submissions}
+    with open(args.tpms_file) as f:
+        reader = csv.reader(f)
+        headers = reader.next()
+        print headers
+        for paper_number, email, _ in reader:
+            paper_id = 'Paper{0}'.format(paper_number)
+            if paper_id in configuration_note.content['assignments']:
+                assignment_entry = configuration_note.content['assignments'][paper_id]
+                assignment_entry['assigned'].append(email)
```

Comment:
why don't we try to get the profile of the reviewer and save the tilde id, is going to be very hard to debug the assignments.